zip-0315.rst

```diff
@@ -1,9 +1,629 @@
+  double-spend not to occur, e.g. TXOs created by the wallet's internal TXO
+  handling.
+* An untrusted TXO is a TXO received by the wallet that is not trusted.
+
+
+Wallets can then require that untrusted TXOs need more confirmations before
+they become spendable than trusted TXOs. This provides an improved trade-off
+between latency on the one hand, and reliability and safety on the other.
+
+
+Specification
+=============
+
+Long-term storage of funds
+--------------------------
+
+It is RECOMMENDED that wallets only hold funds as shielded in the long term;
+that is, if a wallet supports receiving transparent funds (or supports
+importing a seed from another wallet that might have done so), then it SHOULD
+either auto-shield or opportunistically shield such funds by default.
+
+Following this recommendation improves both collective privacy and the user's
+individual privacy, by maximizing the size of the note anonymity set over time.
+
+The remainder of this specification assumes a wallet that follows the above
+recommendation, except where explicitly noted.
+
+A wallet MAY allow users to disable auto-shielding.
+
+
+Trusted and untrusted TXOs
+--------------------------
+
+A wallet SHOULD treat received TXOs that are outputs of transactions created
+by the same wallet, as trusted TXOs. Wallets MAY enable users to mark specific
+external transactions as trusted, allowing their received TXOs also to be
+classified as trusted TXOs.
+
+A wallet SHOULD have a policy that is clearly communicated to the user for
+the number of confirmations needed to spend untrusted and trusted TXOs
+respectively. The following confirmation policy is RECOMMENDED:
+
+* 10 confirmations, for untrusted TXOs;
+* 3 confirmations, for trusted TXOs.
+
+Rationale for the given numbers of confirmations
+''''''''''''''''''''''''''''''''''''''''''''''''
+
+The rationale for choosing three confirmations for trusted TXOs is that
+empirically, reorgs are usually less than three blocks.
+
+The consequences of attempting to spend a trusted TXO may be less severe in the
+case of a rollback than the consequences of attempting to spend an untrusted TXO.
+The value received from a trusted TXO should always be recoverable, whereas
+recovering value received from an untrusted TXO may require the user to request
+that funds are re-sent.
+
+
+Categories of TXOs according to spendability
+--------------------------------------------
+
+A TXO is *spendable*, relative to a given block chain and wallet state,
+if and only if all of the following are true in that state:
+
+* the TXO is unspent;
+* the TXO is not committed to be spent in another transaction created
+  by this wallet; and
+* the wallet has the TXO's spending key (for whatever protocol the TXO uses).
+
+A TXO is *confirmed-spendable*, relative to a given block chain and
+wallet state, if and only if all of the following are true in that state:
+
+* the wallet is synchronized; and
+* the TXO is spendable; and
```

Comment:
This is too strong. It's sufficient that we know that the TXO is spendable, which requires that we know it is unspent at the current chain tip. A condition on knowing that it is unspent at the current chain tip is that we've synchronized the nullifier set to that tip, but that belongs in the definition of "known-spendable".

```suggestion
* the TXO is known-spendable; and
```